### PR TITLE
Release 2.2.0: four dogfooding fixes (TASK-049, 050, 051, 054)

### PR DESCRIPTION
**Pull request by refined**
Four of the six gaps found by dogfooding TaskPlanner in [smekai/isotopy](https://github.com/smekai/isotopy). Cut as **2.2.0** (minor, not patch): this changes how agents behave in other people's repositories.

## Why these four, together

The generated agent instructions regenerate in **every consumer repository** on a version bump — `synchronizeTaskPlannerProject()` compares `taskplannerVersion` and rewrites `AGENTS.md`/`CLAUDE.md`/`.cursorrules`. Three of these four edit that text. Shipping them separately would have churned everyone's instruction files four times for one coherent change.

TASK-052 (blocked-until) and TASK-053 (DONE archiving) stay in the backlog: both are design decisions with one real instance each, and TASK-052 partly dissolves now that epic grouping exists.

## TASK-049 — `aiPlanRequired: false` now genuinely turns planning off

The flag was already honoured across most of the template — that part was fixed after 2.1.4. What survived was one bullet under *Mandatory checklist (do not skip)* that still demanded a plan and sent the reader to go check the config file:

> `- **Plan:** If this project requires a plan (check the **aiPlanRequired** field in .tasks/config.json), the ` `### Plan` ` block must exist in IN_PROGRESS **before** coding...`

The generator knows the answer, so emitting a conditional was the bug. Dropped, along with two incidental mentions in the Done step and the Work Log section.

## TASK-050 — `sortBy` removed from `.tasks/config.json`

Planning turned up more than the reported ambiguity: the field was **dead**. Zero reads anywhere in `src/` — sort order comes from the VS Code setting through `getSortBy()`, and `setSortBy()` writes back there. It existed only in the type and the default, was written into every config on init, and sitting beside `insertPosition` read as a promise about file layout. isotopy's agent rewrote `BACKLOG.md` to satisfy it and corrupted the encoding doing so.

Removed, with a v3 migration that strips the key from existing configs and leaves unknown keys alone. Generated instructions now state that order within a file carries no meaning — removing the field alone would have relocated the question, since `insertPosition` still lives there and genuinely does affect layout.

## TASK-051 — `epic` is no longer write-only

Added to the `taskplanner_create` and `taskplanner_update` inputs, and to `groupBy`, following the `assignee` precedent exactly (`entry.task.epic || 'No epic'`). Also surfaced in the MCP text output, which had omitted it even when set.

Verified end to end against a real server on a scratch board: create with epic → move → get → update, surviving every step and landing in the markdown as `**Epic:** 2.3.x`.

## TASK-054 — Initialize can write `.mcp.json`

Two decisions worth flagging:

**`npx`, not a resolved path.** `.mcp.json` is committed and shared across a team's machines, so anything install-specific breaks for every other clone. The written entry is `{ "command": "npx", "args": ["-y", "@smekai/taskplanner"] }` with no `TASKPLANNER_WORKSPACE_ROOT` — that would need an absolute path too, and the host runs with the repo as cwd while the instructions already tell agents to pass `workspace_root` per call.

**Asked, not silent.** Unlike `AGENTS.md`, this file tells an agent what to execute, so it is never a side effect of Initialize. The answer is remembered in `config.json`; the Setup menu writes it later for anyone who declined.

Merging is real merging: an existing `.mcp.json` keeps its other servers and unknown top-level fields, and a file that cannot be parsed is left untouched rather than overwritten. All four paths verified on scratch repositories.

## Verification

- `npm run release:check` green: **143 tests** (up from 133), all version markers at 2.2.0, plugin validators and the published-artifact smoke test pass.
- End-to-end runs against a real MCP server for the epic round-trip and against scratch repositories for every `.mcp.json` path.
- This repository's own `CLAUDE.md`/`AGENTS.md`/`.cursorrules` regenerated once at the end, so the changes are dogfooded rather than only shipped.

## Notes

- **No `.mcp.json` in this repository yet** — `npx -y @smekai/taskplanner` cannot resolve until the package is published. Worth adding immediately after the first publish; it is the last piece of "the defect reproduces on its own maintainer".
- Committed with `--no-verify` after `npm run validate:versions` passed, per `CONTRIBUTING.md`, so the pre-commit hook did not bump 2.2.0 to 2.2.1.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by refined — COMMENTED** [review]
Reviewed the four behavior changes and ran the complete release gate locally: all 143 tests, validators, build, and published-package smoke checks pass. I found three functional edge cases not covered by the current suite.

File: src/extension/views/webview/taskListPanel.ts [hunk 1]
```diff
@@ -396,6 +396,7 @@ export class TaskListViewProvider implements vscode.WebviewViewProvider {
     const groupByItems = [
       { value: 'status', label: 'Status' },
       { value: 'assignee', label: 'Assignee' },
+      { value: 'epic', label: 'Epic' },
```
Comment (refined) [inline src/extension/views/webview/taskListPanel.ts:399]: [P1] Add `epic` to the persisted `GroupBy` contract as well. `extensionConfig.ts` still declares `GroupBy` as `status | assignee | date | none`, and `GROUP_BY_VALUES` omits `epic`. Selecting this item writes `epic`, but the configuration-change listener refreshes the view and `getGroupBy()` rejects that saved value and falls back to `status`, so the new grouping cannot stay selected. Please add `epic` to both the type and accepted-value list (ideally deriving one from the other) and cover the settings round-trip.

File: src/extension/commands/setup.ts [hunk 2]
```diff
@@ -42,6 +44,12 @@ export function registerSetupCommand(
         action: 'initAi',
       });
 
+      items.push({
+        label: `$(plug) Write ${MCP_CONFIG_FILE}`,
```
Comment (refined) [inline src/extension/commands/setup.ts]: [P1] Do not offer this action before the board is initialized. This item is pushed even when `isInitialized` is false; choosing it calls `writeMcpConfigNow()`, whose `configManager.save()` creates `.tasks/` containing only `config.json`. Every initialization check uses `existsSync(tasksDir)`, so after reload the project is treated as initialized and the Initialize item disappears even though all state files are absent and the MCP server has no usable board. Gate this item on `isInitialized`, or avoid persisting the choice into `.tasks` until initialization completes.

File: src/core/config/configManager.ts [hunk 3]
```diff
@@ -42,8 +42,16 @@ export class ConfigManager {
...
 
+    // v3: Drop "sortBy". It was never read — sort order is a view setting, not project layout —
+    // and sitting beside insertPosition it read as a file-ordering contract.
+    const legacy = this.config as TaskPlannerConfig & { sortBy?: unknown };
+    if (legacy.sortBy !== undefined) {
+      delete legacy.sortBy;
+      changed = true;
+    }
+
     if (changed) {
```
Comment (refined) [inline src/core/config/configManager.ts:64]: [P2] Advance the schema based on the stored version, not only on whether this migration happened to delete a field. A v2 config that already lacks `sortBy` and already contains Rejected leaves `changed` false, so `load()` returns version 2 and never updates the file; I reproduced `{ loadedVersion: 2, onDiskVersion: 2 }` with that shape. Mark versions below 3 as changed (while avoiding downgrading future versions) so `version` reliably records that the v3 migration has been applied.

File: src/extension/views/webview/taskListPanel.ts (hunk 1 above)
Comment (refined) [reply src/extension/views/webview/taskListPanel.ts:399]: Fixed in 5ab011a. Confirmed exactly as described: `GroupBy` and `GROUP_BY_VALUES` in `extensionConfig.ts` both omitted `epic`, so `getGroupBy()` rejected the saved value and fell back to `status` — the grouping could be selected but not kept.

Took the deriving suggestion further than one file. The vocabulary now lives once in `src/core/model/messages.ts`:

```ts
export const GROUP_BY_VALUES = ['status', 'assignee', 'epic', 'date', 'none'] as const;
export type GroupBy = (typeof GROUP_BY_VALUES)[number];
```

`taskFilter.ts`, the `TaskFilter` message type and `extensionConfig.ts` all derive from it, so four of the five former declarations are gone rather than kept in sync by hand.

The two that cannot be collapsed — the `taskplanner.groupBy` enum in package.json and the panel's `groupByItems` menu — are covered by tests asserting they match the core list. Negative-tested both: removing `epic` from either fails with `expected [ Array(4) ] to deeply equal [ 'assignee', 'date', 'epic', …(2) ]`. A third test runs `groupTasks` for every declared value so a new grouping cannot be added without a branch.

File: src/extension/commands/setup.ts (hunk 2 above)
Comment (refined) [reply src/extension/commands/setup.ts]: Fixed in 5ab011a. Correct, and the consequence is worse than a stray directory: `save()` does `mkdirSync(this.tasksDir)`, so choosing this on an uninitialized project leaves `.tasks/` holding only `config.json`, and since every check is `existsSync(tasksDir)` the project then reads as initialized with no state files at all — the Initialize entry disappears and the board is unusable.

Gated the item on `isInitialized`, with a comment recording why, since the reason is not visible from the call site. The Initialize path is unaffected: `promptForMcpConfig` runs after the state files are written.

File: src/core/config/configManager.ts (hunk 3 above)
Comment (refined) [reply src/core/config/configManager.ts:64]: Fixed in 5ab011a. Reproduced your exact numbers before changing anything — a v2 config already lacking `sortBy` and already holding `Rejected` gave `{ loadedVersion: 2, onDiskVersion: 2 }`, so `version` recorded nothing about whether the v3 migration had run.

The version is now set from the stored value rather than from whether a field happened to be deleted, against a `CONFIG_SCHEMA_VERSION` constant, and the `changed` flag drives the save as before:

```ts
if (this.config.version < CONFIG_SCHEMA_VERSION) {
  this.config.version = CONFIG_SCHEMA_VERSION;
  changed = true;
}
```

Two tests: the no-op v2 case now reaches 3 both in memory and on disk, and a config claiming version 99 is left at 99 rather than downgraded.
